[wilsonh0] iP

wilsonh0 opened the pull request:
# 🐢 Oogway 
> "Yesterday is history, tomorrow is a mystery, but today is a gift. That is why it is called the present." — Master Oogway

### Oogway frees your mind from the burden of remembering tasks. It is:
- 🐢 **Calm and simple** – _No distractions, just your tasks._
- 📜 **Easy to follow** – _Clear and intuitive._
- ⚡ **Swift, yet steady** – _Focus on what matters._

#### 🍃 Features
- [X] Tasks management
- [X] Deadline management
- [ ] Reminders (coming soon)

#### 🌸 Using Oogway
1. Download the latest release [here](https://github.com/wilsonh0/ip).
2. Double click the jar file.
3. Add tasks, and let Oogway manage it for you.


#### 📜 Code Snippet
```java
public static void main(String[] args) {
    new Oogway(null).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

wilsonh0 committed:
Rename main class and add initial skeleton for Oogway bot

wilsonh0 committed:
Add support for echo

wilsonh0 committed:
Add ability to store text and display list of stored items

wilsonh0 committed:
Add Task class to represent tasks with description and status

wilsonh0 committed:
Add ability to mark tasks as done or undone

wilsonh0 committed:
Add input validation for task creation and enhance code clarity

wilsonh0 committed:
Add Task subclasses (Deadline, Event, ToDo) to support different task types

wilsonh0 committed:
feat: Add support for tracking ToDo, Event, Deadline tasks

refactor: Replaced horizontalLine() with wrapMessage() for flexibility

wilsonh0 committed:
test: Add automated Text UI Testing

wilsonh0 committed:
feat: Add OogwayException for custom error management

wilsonh0 committed:
refactor: Use OogwayException for consistent error handling

wilsonh0 committed:
feat: Add support for deleting tasks from list

wilsonh0 committed:
docs: Update Javadoc comments for methods

wilsonh0 committed:
feat: Update Task and its subclasses to support file saving

wilsonh0 committed:
feat: Add FileHandler and Storage classes for file saving and improved OOP design

wilsonh0 committed:
feat: Enhance Oogway with file handling and storage integration

wilsonh0 committed:
chore: Add data/save.txt to .gitignore

wilsonh0 committed:
feat: Add datetime parsing to Deadline and Event classes

wilsonh0 committed:
refactor: Update Oogway to handle new exceptions and support save file loading

wilsonh0 committed:
refactor: Move UI handling into Ui class

wilsonh0 committed:
refactor: Move UI logic from FileHandler into Ui class

wilsonh0 committed:
refactor: Replace direct UI handling in Oogway with Ui class

wilsonh0 committed:
refactor: Organize classes into packages

wilsonh0 committed:
Merge branch 'add-gradle-support'

wilsonh0 committed:
feat: Update gradle config

wilsonh0 committed:
refactor: Rename Storage and FileHandler class for better clarity

Storage was originally a wrapper around an ArrayList of tasks, which
is more accurately represented as TaskList.

FileHandler, responsible for saving/loading tasks, has been renamed
to Storage to better reflect its role.

wilsonh0 committed:
feat: Add abstract Command class and its subclasses

wilsonh0 committed:
feat: Add Parser class for user input processing

wilsonh0 committed:
refactor: Update Oogway to use Parser for input handling

wilsonh0 committed:
test: Add JUnit tests for Parser and Todo

wilsonh0 committed:
docs: Add Javadocs to Oogway, Task, Ui classes

wilsonh0 committed:
docs: Add Javadocs to Storage and TaskList classes

wilsonh0 committed:
docs: Add Javadocs to Parser class

wilsonh0 committed:
feat: Add support for searching tasks by keyword in description

wilsonh0 committed:
style: Apply coding standard improvements across multiple files

wilsonh0 committed:
merge: Resolve conflicts from 'branch-A-JavaDoc'

wilsonh0 committed:
merge: Resolve conflicts from 'branch-A-CodingStandard'

wilsonh0 committed:
fix: Resolve missed merge conflicts from last merge

wilsonh0 committed:
refactor: Remove generic type <T> from Command

Ben926 commented on src/main/java/oogway/Oogway.java:
```diff
@@ -0,0 +1,95 @@
+ * The main class for the Oogway application. It manages user interaction, command execution,
+ * and task storage.
+ */
+public class Oogway {
```
I like how your description for this class is very straightforward yet detailed.

Ben926 replied:
I like how your description for this class is very straightforward yet detailed.

Ben926 commented on src/main/java/oogway/Oogway.java:
```diff
@@ -0,0 +1,95 @@
+     */
+    public Oogway(String filePath) {
+        ui = new Ui();
+
```
I like that you separate logic blocks by a single line, as stated in the coding standard

Ben926 replied:
I like that you separate logic blocks by a single line, as stated in the coding standard

Ben926 commented on src/main/java/oogway/commands/AddTaskCommand.java:
```diff
@@ -0,0 +1,77 @@
+import oogway.tasks.Event;
+import oogway.tasks.Task;
+import oogway.tasks.ToDo;
+
```
Should you add a description header for this public class?

Ben926 replied:
Should you add a description header for this public class? I noticed some of your public classes are missing that.

Ben926 commented on src/main/java/oogway/parser/Parser.java:
```diff
@@ -0,0 +1,146 @@
+    private static final Pattern TODO_PATTERN = Pattern.compile("^(?<description>[^/]+)$");
+
+    private static final String MESSAGE_INVALID_COMMAND = "Command not found!";
+    private static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
```
I like that you followed the variable name convention for constant variables!

Ben926 replied:
I like that you followed the variable name convention for constant variables!

Ben926 commented on src/main/java/oogway/parser/Parser.java:
```diff
@@ -0,0 +1,146 @@
+    public Command parseTodo(String arguments) {
+        final Matcher matcher = TODO_PATTERN.matcher(arguments.trim());
+        if (!matcher.matches()) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, "Usage: todo <description>"));
```
Should the second line of this long statement be indented 8 spaces right of the first line?

Ben926 reviewed (commented):
Overall, I thought the code was easy to read and detailed! Some minor coding standard improvements can be made here and there as mentioned, although there were many good practices used too!

wilsonh0 committed:
chore: Add checkstyle for code standard enforcement

Engulfy commented on src/main/java/oogway/commands/AddTaskCommand.java:
```diff
@@ -0,0 +1,77 @@
+    private final String to;
+
+    // For Event
+    public AddTaskCommand(TaskList taskList, String taskType, String taskDescription,
```
I like how the class name is clear and easy to understand, with comment of which constructor is for which kind of task

Engulfy commented on src/main/java/oogway/tasks/Event.java:
```diff
@@ -0,0 +1,97 @@
+    protected LocalDateTime startDateTime;
+    protected LocalDateTime endDateTime;
+
+    protected DateTimeFormatter startDateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
```
I like how you named the four different dates separately and it is very clear

Engulfy commented on src/main/java/oogway/commands/FindTaskCommand.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    public CommandResult<List<Task>> execute() {
+        List<Task> foundTasks = taskList.findTasks(keyword);
```
I like how the name of the variable corresponds to what the method is doing

Engulfy commented on src/main/java/oogway/commands/FindTaskCommand.java:
```diff
@@ -0,0 +1,33 @@
+        }
+
+        StringBuilder message = new StringBuilder("Here are the matching tasks in your list:\n");
+        for (int i = 0; i < foundTasks.size(); i++) {
```
Perhaps can change the i to index for it to be more readable?

Engulfy reviewed (commented):
Very interesting project name and creativity! Everything looks good to me, keep up the good work!

wilsonh0 committed:
feat: Add JavaFX GUI while retaining CLI functionality

wilsonh0 committed:
style: Fix Checkstyle violations across multiple files

The code contains inconsistent indentation, improper naming
conventions, unused imports, and misformatted JavaDocs.

These issues reduce readability and make maintenance harder.

Fix indentation, enforce naming conventions,
remove unused imports, and correct JavaDocs.

This improves code consistency without changing functionality.

wilsonh0 committed:
refactor: improve Parser readability and consistency

Parser currently has inline regex patterns and inconsistent method
access. This makes future modifications harder and reduces code
clarity.

Extract regex patterns into named constants (`DESC`, `FROM`,
`TO`, `BY`) and make helper methods consistently `private`.

This keeps the code cleaner and more maintainable without changing
behavior.

wilsonh0 committed:
feat: Add TaskType enum for better task handling

Replaced string-based task types with TaskType enum to prevent typos
and ensure valid task types.

This makes the code clearer and easier to maintain.

wilsonh0 committed:
refactor: update AddTaskCommand to use TaskType and factory methods

Replaced overloaded constructors with factory methods
(`createTodo()`, `createDeadline()`, `createEvent()`) for cleaner
task creation.

Updated AddTaskCommand to use TaskType enum instead of raw strings,
improving type safety and readability.

wilsonh0 committed:
refactor: update Parser to use AddTaskCommand factory methods

Modified Parser to use `createTodo()`, `createDeadline()`, and
`createEvent()` instead of direct constructor calls.

This ensures consistent task creation and improves readability.

wilsonh0 committed:
refactor: Add assertions for better debugging and code safety

wilsonh0 committed:
Merge pull request #1 from wilsonh0/branch-A-CodeQuality

Fix Checkstyle violations and improve Parser code quality

wilsonh0 committed:
Merge pull request #2 from wilsonh0/branch-A-Assertions

Add assertions for better debugging and code safety

wilsonh0 committed:
ci: Add GitHub Actions CI workflow for Gradle build and tests

wilsonh0 committed:
feat: Add sorting of tasks by datetime

wilsonh0 committed:
feat: Add help command to list all available commands

wilsonh0 committed:
Merge pull request #3 from wilsonh0/branch-C-Sort

Add sorting of tasks by datetime

wilsonh0 committed:
ui: Improve styling, colors, and add title/icon

wilsonh0 committed:
style: Improve formatting for better readability

wilsonh0 committed:
fix: Resolve path mismatches by normalizing custom file path

wilsonh0 committed:
docs: Add Javadocs and cite external code contributions

wilsonh0 committed:
docs: Add attribution comments in Javadoc

wilsonh0 committed:
fix: Add missing suer dialog box on program exit

wilsonh0 committed:
ui: Adjust program window width and picture size

wilsonh0 committed:
docs: Add user guide and Ui.png

wilsonh0 committed:
docs: Update README.md

wilsonh0 committed:
Update README.md

wilsonh0 committed:
Update README.md
